src/engine/Story.ts

```diff
@@ -1852,7 +1990,12 @@ export class Story extends InkObject {
 
       for (let varName of keys) {
         let observers = this._variableObservers.get(varName)!;
-        observers.splice(observers.indexOf(observer), 1);
+
+        if (observers !== null) {
+          observers.splice(observers.indexOf(observer), 1);
+        } else {
+          this._variableObservers.delete(specificVariableName);
```

Comment:
Unless I'm missing something we have the same issue here as above: after removing the observer from this._variableObservers[varName], we don't call this._variableObservers.delete() even of that was the last of the observers.